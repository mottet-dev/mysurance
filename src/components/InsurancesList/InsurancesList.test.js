import React from 'react';
import { shallow } from 'enzyme';
import InsuranceList from './InsurancesList';
import { List, ListItem } from 'react-native-elements';

describe('<InsurancesList />', () => {
    const component = shallow(<InsuranceList />);

    it('renders properly', () => {
        expect(component).toMatchSnapshot();
    });

    it('containes 1 <List />', () => {
        expect(component.find(List).length).toEqual(1);
    });

    it('containes at 0 or more <ListItem />', () => {
        expect(component.find('View > List').children(ListItem).length).toBeGreaterThanOrEqual(0);
    });
});