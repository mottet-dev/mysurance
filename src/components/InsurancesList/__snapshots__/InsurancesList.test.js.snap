// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InsurancesList /> renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InsurancesList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ListViewMock
        dataSource={
          ListViewDataSource {
            "items": 12,
          }
        }
        renderRow={[Function]}
        renderScrollComponent={[Function]}
      />,
      "containerStyle": Object {
        "backgroundColor": "rgba(0,0,0,0)",
        "marginTop": 0,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "dataSource": ListViewDataSource {
          "items": 12,
        },
        "renderRow": [Function],
        "renderScrollComponent": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ListViewMock
          dataSource={
            ListViewDataSource {
              "items": 12,
            }
          }
          renderRow={[Function]}
          renderScrollComponent={[Function]}
        />,
        "containerStyle": Object {
          "backgroundColor": "rgba(0,0,0,0)",
          "marginTop": 0,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "dataSource": ListViewDataSource {
            "items": 12,
          },
          "renderRow": [Function],
          "renderScrollComponent": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
